File: app/src/main/java/dev/ragnarok/fenrir/api/HttpLogger.java
package dev.ragnarok.fenrir.api;

import dev.ragnarok.fenrir.Constants;
import okhttp3.logging.HttpLoggingInterceptor;


public class HttpLogger {

    public static final HttpLoggingInterceptor DEFAULT_LOGGING_INTERCEPTOR = new HttpLoggingInterceptor();

    static {
        if (Constants.IS_DEBUG) {
            DEFAULT_LOGGING_INTERCEPTOR.setLevel(HttpLoggingInterceptor.Level.BODY);
        } else {
            DEFAULT_LOGGING_INTERCEPTOR.setLevel(HttpLoggingInterceptor.Level.NONE);
        }
    }
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/api/HttpLogger.kt
package dev.ragnarok.fenrir.api

import dev.ragnarok.fenrir.Constants
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor

object HttpLogger {
    val DEFAULT_LOGGING_INTERCEPTOR: HttpLoggingInterceptor by lazy {
        HttpLoggingInterceptor().setLevel(if (Constants.IS_DEBUG) HttpLoggingInterceptor.Level.BODY else HttpLoggingInterceptor.Level.NONE)
    }

    val UPLOAD_LOGGING_INTERCEPTOR: HttpLoggingInterceptor by lazy {
        HttpLoggingInterceptor().setLevel(if (Constants.IS_DEBUG) HttpLoggingInterceptor.Level.HEADERS else HttpLoggingInterceptor.Level.NONE)
    }

    fun adjust(builder: OkHttpClient.Builder) {
        if (Constants.IS_DEBUG) {
            builder.addInterceptor(DEFAULT_LOGGING_INTERCEPTOR)
        }
    }

    fun adjustUpload(builder: OkHttpClient.Builder) {
        if (Constants.IS_DEBUG) {
            builder.addInterceptor(UPLOAD_LOGGING_INTERCEPTOR)
        }
    }
}